spec/database_cleaner/configuration_spec.rb

```diff
@@ -93,6 +86,9 @@ def stub_cleaners(array)
     end
 
     it "should retrieve a db rather than create a new one" do
+      stub_const "DatabaseCleaner::ActiveRecord", Module.new
+      stub_const "DatabaseCleaner::ActiveRecord::Truncation", Class.new
```

Comment:
Hmm, I having a hard time getting this to work with `object_double`, but I may be doing it wrong! Can you say a bit more about how this might work?

Reply:
Sorry I meant `class_double`.

Reply:
Oh, okay! But I can't seem to get that working either :rofl:

Can you say a bit more about how `class_double` could improve this spec? Sorry I'm so dense... I used to think I was pretty darn knowledgeable about RSpec!

Reply:
`class_double("DatabaseCleaner::ActiveRecord::Truncation").as_stubbed_const` gives you a double which will only allow you to stub/expect methods that exist on the class, its a verifying double :)

Reply:
Ohhh, I see what you're getting at. I actually don't think that would be helpful here, because the `database_cleaner-active_record` gem isn't loaded at all! This suite is testing core in isolation from any of the adapters. That's why we need to stub these constants so that the strategy lookup code has something to find when we say `DatabaseCleaner[:active_record].strategy = :truncation`.

So, I don't think a verifying double would add much value, because the only methods being called on these constants is `const_get` on DC::AR to get DC::AR::Truncation, and `.new` on DC::AR::Truncation.

Now that I write this, it occurs to me that this situation could probably be improved by using an obviously fake set of constants, so that its clear that the implementation doesn't matter.

What do you think?

Reply:
They behave like normal doubles if the class isn't loaded, they offer an extra level of protection when they are.

Reply:
Wow, once again, I didn't know that! Very cool. I'm convinced. I'll give it a shot!

Reply:
Okay, pushed a new commit using `class_double` that I hope is making good use of it. Thanks for being a great teacher!